pyocd/flash/loader.py

```diff
@@ -122,7 +122,11 @@ def program(self, file_or_path, format=None, **kwargs):
         try:
             # Open the file if a path was provided.
             if isPath:
-                file_obj = open(file_or_path, "rb")
+                mode = 'rb'
+                if format == 'hex':
```

Comment:
Using `format` is not a good idea in Python since it's already used by a built-in function. I would suggest (but not in this PR) to use a different name, e.g `file_type` or whatever.

Reply:
Ah, good point! I was just reusing existing code with that variable name. Todo note taken.